File: apps/api/src/main/kotlin/com/wewiins/saas_api/interfaces/Statistic.kt
package com.wewiins.saas_api.interfaces

data class Statistic(
    val totalRevenue: ComparisonStat<Double>,
    val totalCharges: ComparisonStat<Double>,
    val totalBooking: ComparisonStat<Int>,
    val averageOccupancy: ComparisonStat<Int>,
    val averageCancellation: ComparisonStat<Double>,
    val averageParticipants: ComparisonStat<Double>,
    val totalFavorites: ComparisonStat<Int>,
    val averageScore: ComparisonStat<Double>,
    val visitsByPeriod: List<VisitDataPoint>,
    val scoreDistribution: List<ScoreDistribution>,
    val filterRangeDays: Int,
)

data class VisitDataPoint(
    val date: Long,
    val count: Int
)

data class ScoreDistribution(
    val star: Int,
    val count: Int
)

File: apps/client/src/app/dto/statistic.ts
import { IStatComparisonDto } from "./common";

export interface IStatisticDto {
    totalRevenue: IStatComparisonDto<number>;
    totalCharges: IStatComparisonDto<number>;
    totalBooking: IStatComparisonDto<number>;
    averageOccupancy: IStatComparisonDto<number>;
    averageCancellation: IStatComparisonDto<number>;
    averageParticipants: IStatComparisonDto<number>;
    totalFavorites: IStatComparisonDto<number>;
    averageScore: IStatComparisonDto<number>;
    visitsByPeriod: IVisitDataPointDto[];
    scoreDistribution: IScoreDistributionDto[];
    filterRangeDays: number;
}

export interface IVisitDataPointDto {
    date: number;
    count: number;
}

export interface IScoreDistributionDto {
    star: number;
    count: number;
}